extensions/picocli/deployment/src/main/java/io/quarkus/picocli/deployment/PicocliProcessor.java

```diff
@@ -44,6 +47,14 @@ IndexDependencyBuildItem picocliIndexDependency() {
         return new IndexDependencyBuildItem("info.picocli", "picocli");
     }
 
+    @BuildStep
+    void configBuilder(BuildProducer<StaticInitConfigBuilderBuildItem> staticInitConfigBuilder,
+            BuildProducer<RunTimeConfigBuilderBuildItem> runTimeConfigBuilder) {
+        System.out.println("PicocliProcessor.configBuilder");
+        staticInitConfigBuilder.produce(new StaticInitConfigBuilderBuildItem(PicocliConfigBuilder.class.getName()));
+        runTimeConfigBuilder.produce(new RunTimeConfigBuilderBuildItem(PicocliConfigBuilder.class.getName()));
+    }
+
```

Comment:
YAY!!! Due to nature of build steps, it should be possible to use build time config (read by the build step) to determine whether or not this should happen (so it isn't a breaking/surprising change).

What if we started with: 
```
quarkus.picocli.static.config-builder.enabled=true (default false)
quarkus.picocli.runtime.config-builder.enabled=true (default false)
```

We could then later add :
```
quarkus.picocli.static.config-builder.class-name=MyCustomStaticBuilder
quarkus.picocli.runtime.config-builder.class-name=MyCustomRuntimeBuilder
```
(e.g. eventual answer to #19128)

Reply:
We can expose them via the ServiceLoader instead of config. Might be a little bit cleaner.

Yes, regarding the builder in the picocli extension, it was just to try the API and see if it actually worked. I can remove it and let you implement it as you see fit :)

Reply:
Well. having this in the Picocli extension is good (see #19128) .. I don't want command line application creators to be obligated to write their own extension.. I can't tell if I prefer the service loader or the config attribute..  I feel like ServiceLoaders are nasty.. because you can't then turn them off or remove them later.

Reply:
Not exactly. I would read the service loader files during build time and register the classes directly with the builder (like we do with other config APIs). In essence, the consumer would use the service loader mechanism via `META-INF/services` to register the builder, but they would be read and registered directly in build time and would not be part of the final jar.

Reply:
Here is an example:
https://github.com/quarkusio/quarkus/blob/main/core/deployment/src/main/java/io/quarkus/deployment/steps/ConfigGenerationBuildStep.java#L105-L107
https://github.com/quarkusio/quarkus/blob/main/core/deployment/src/main/java/io/quarkus/deployment/steps/ConfigGenerationBuildStep.java#L105-L107

Reply:
y. I get it. I just find service loader awkward.

Reply:
It is just the registration mechanism. I think it would align better with the rest of the API. Everything else in Config uses it (sources, converters, interceptors, factories, etc.). 

One of the issues to prefer this approach is that it may be more awkward to use configuration to configure the config instance itself. In this case, it works, because the proposed configuration is build time and only applied to runtime, but if we were in the same phase it won't work because it creates a chicken / egg problem.

What do you think?

Reply:
For the moment I added the configuration `quarkus.picocli.disable-default-config-sources` to disable the default sources. By default, this is turned off, so it shouldn't break current users if they are relying on this.

Reply:
BTW, this won't completely solve #19549, because we need to have the flag turned on on the `quarkus` cli, or the same issue will occur. @ebullient what do you want to do in this case?

Reply:
I will make/test the updates to the quarkus cli .. don't worry about that. This has to exist first. ;)